### PR TITLE
chore: create module directory for discovery

### LINKED ISSUE #19: create directory & readme file for discovery.js module
i think we're going to follow a pattern of each big module in the lib folder living in a directory with a README.md file, so we can start doing that with discovery.js

tasks:
- move lib/discovery.js to lib/discovery/index.js
- decide what these internal readme files should include (basic description, usage example, link to tests, what else?)
- create README.md file with content

### PR DESCRIPTION
**Pull request by achou11**
Closes #19 

The structure is in place but need some assistance with filling out the finer details, meaning about 80% of the readme 😄

### CODE REVIEW

File: lib/discovery/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Discovery
+
+> Discover other peers via the DHT or MDNS and replicate with them
+
+## Usage
+
+The `Discovery` class provides an abstraction layer that allows peer discovery using either a [distributed hash table](https://en.wikipedia.org/wiki/Distributed_hash_table) (DHT) or [Multicast DNS](https://en.wikipedia.org/wiki/Multicast_DNS) (mDNS).
```
Comment (sethvincent) [inline lib/discovery/README.md:7]:
```suggestion
The `Discovery` class provides an abstraction layer that allows peer discovery using either a [distributed hash table](https://en.wikipedia.org/wiki/Distributed_hash_table) (DHT) via [hyperswarm](https://npmjs.com/hyperswarm) or [Multicast DNS](https://en.wikipedia.org/wiki/Multicast_DNS) (mDNS) via [multicast-service-discovery](https://npmjs.com/multicast-service-discovery).
```

File: lib/discovery/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
```
Comment (sethvincent) [inline lib/discovery/README.md:8]: We could add a snippet like this:

```js
import Hypercore from 'hypercore'
import { KeyManager } from '@mapeo/crypto'
import { Discovery } from './index.js'

const rootKey = KeyManager.generateRootKey()
const keyManager = new KeyManager(rootKey)
const identityKeyPair = keyManager.getIdentityKeypair()

const core = new Hypercore()
await core.ready()

const discover = new Discovery({
	identityKeyPair,
	mdns: true,
	dht: true,
})

await discover.ready()

discover.on('connection', async (connection, peer) => {
	core.replicate(connection)
})

await discover.join(core.discoveryKey)
```

This is likely the simplest possible usage of the discovery module.

Comment (achou11) [reply lib/discovery/README.md:35]: Addressed via 179ba173c7c1d0f8212bb60e52d4a7b71508eb84

**Comment by sethvincent** [comment]
cool, looks good to me!

**Pull request merged**